### PR TITLE
Add cost per KWh to charges dashboard.

### PR DESCRIPTION
**Pull request by jyothri**
Added a new column to show how economic the charges at the location has been. 
Not all sessions are equal i.e some charge by KWh, some by time. Additionally there could be a flat session charge.

### CODE REVIEW

**Comment by jyothri** [comment]
Hi @adriankumpf 
Wanted to check if I can get an estimate of when this could be addressed?

Thanks,
Jyothrilinga.

File: grafana/dashboards/charges.json [hunk 1]
```diff
@@ -741,7 +757,7 @@
...
...at=', latitude, '&lng=', longitude)\n         WHEN geofence_id IS NOT NULL THEN CONCAT(geofence_id, '/edit')\n    END as path,\n    car_id,\n    id,\n    -- Columns\n    start_date,\n    end_date,\n    COALESCE(geofence_name, address) as address,    \n    duration_min,\n    cost,\n    cost/NULLIF(charge_energy_added, 0)  as cost_per_kwh,\n    charge_energy_added,\n    charge_energy_used,\n    CASE WHEN charge_energy_used IS NULL THEN NULL ELSE LEAST(charge_energy_added / NULLIF(charge_energy_used, 0), 1.0) END as charging_efficiency,\n    convert_celsius(outside_temp_avg, '$temp_unit') AS outside_temp_avg_$temp_unit,\n    charge_energy_added * 60 / NULLIF (duration_min, 0) AS charge_energy_added_per_hour,\n    convert_km((end_[[preferred_range]]_range_km - start_[[preferred_range]]_range_km) * 60 / NULLIF (duration_min, 0), '$length_unit') AS range_added_per_hour_$length_unit,\n    start_battery_level,\n    end_battery_level,\n    convert_km(distance::numeric, '$length_unit') AS distance_$length_unit\n FROM\n    data\nWHERE\n    (distance >= 0 OR distance IS NULL)\n    AND max_charger_voltage >= '$voltage'\n    AND duration_min >= '$min_duration_min'\nORDER BY\n  start_date DESC;",
```
Comment (adriankumpf) [inline grafana/dashboards/charges.json:760]: TeslaMate calculates the total cost using `charge_energy_added` or `charge_energy_used`, whichever is higher.  This should be reflected here too:

```sql    
cost/NULLIF(greatest(charge_energy_added, charge_energy_used), 0)  as cost_per_kwh
```

File: grafana/dashboards/charges.json [hunk 2]
```diff
@@ -706,6 +706,22 @@
                 "value": true
               }
             ]
+          },
+          {
+            "matcher": {
+              "id": "byName",
+              "options": "cost_per_kwh"
+            },
+            "properties": [
```
Comment (adriankumpf) [inline grafana/dashboards/charges.json:715]: A fixed column width would be neat. 130 should be enough.

**Review by adriankumpf — CHANGES_REQUESTED** [review]

**Comment by adriankumpf** [comment]
Superseded by #2693. Thank you anyway!

**Pull request closed without merging**